docs/rest_api/authentication.rst

```diff
@@ -0,0 +1,289 @@
+
+.. code-block:: console
+
+    curl -X POST \
+         -H "Content-Type: application/x-www-form-urlencoded" \
+         -d "grant_type=authorization_code&client_id=CLIENT_ID&redirect_uri=https%3A%2F%2Fexample.com%2Fyour-callback&client_secret=CLIENT_SECRET&code=UNIQUE_CODE_STRING" \
+         https://mautic.example.com/oauth/v2/token
+
+The response returned is a JSON encoded string:
+
+.. code-block:: json
+
+    {
+        "access_token": "ACCESS_TOKEN",
+        "expires_in": 3600,
+        "token_type": "bearer",
+        "scope": "",
+        "refresh_token": "REFRESH_TOKEN"
+    }
+
+Please store this data in a secure location and use it to authenticate API requests.
+
+Refreshing tokens
+^^^^^^^^^^^^^^^^^
+
+The response's ``expires_in`` is the number of seconds the access token is good for and may differ based on what you configured in Mautic. The code handling the authorization process should generate an expiration timestamp based on that value. For example ``<?php $expiration = time() + $response['expires_in']; ?>``. If the access token has expired, you can use the ``refresh_token`` to obtain a new access token.
+
+By default, the refresh token is valid for 14 days unless configured otherwise in Mautic.
+
+* If your app requests a new access token using the refresh token within 14 days, there's no need for any User interaction. Your app gets both a new access token and a new refresh token, which is valid for another 14 days after it's issued;
+* If your app doesn't request a new token using the refresh token within 14 days, you'll need to start from Step One again and redirect the User to Mautic's login.
+
+The refresh token's expiration time is configurable through Mautic's Configuration. 
+
+.. note::
+    The app should monitor for a ``400 Bad Request`` response when requesting a new access token and redirect the User back through the authorization process if that happens.
+
+To obtain a new access token, you should do a POST call to the access token's endpoint ``oauth/v2/token`` using the ``refresh_token`` grant type, like so:
+
+.. code-block:: console
+
+    curl -X POST \
+         -H "Content-Type: application/x-www-form-urlencoded" \
+         -d "grant_type=refresh_token&client_id=CLIENT_ID&client_secret=CLIENT_SECRET&refresh_token=REFRESH_TOKEN" \
+         https://mautic.example.com/oauth/v2/token
+
+The response returned should be a JSON encoded string:
+
+.. code-block:: json
+
+    {
+        "access_token": "NEW_ACCESS_TOKEN",
+        "expires_in": 3600,
+        "token_type": "bearer",
+        "scope": "",
+        "refresh_token": "REFRESH_TOKEN"
+    }
+
+
+Client Credentials flow
+=======================
+
+Using Mautic's API library for the Client Credentials flow
+----------------------------------------------------------
+
+.. warning:: 
+
+    Mautic's API library doesn't have support yet for this flow, but there's an open PR that adds support: https://github.com/mautic/api-library/pull/269
```

Comment:
Please convert this to a link macro using the `make link` command, and commit the link file.